cmd/queue/main.go

```diff
@@ -247,38 +249,68 @@ func handler(reqChan chan queue.ReqEvent, breaker *queue.Breaker, handler http.H
 }
 
 // Sets up /health and /wait-for-drain endpoints.
-func createAdminHandlers() *http.ServeMux {
+func createAdminHandlers(p *readiness.Probe) *http.ServeMux {
 	mux := http.NewServeMux()
-	mux.HandleFunc(requestQueueHealthPath, healthState.HealthHandler(probeUserContainer))
+
+	if p.IsStandardProbe() {
+		mux.HandleFunc(requestQueueHealthPath, healthState.HealthHandler(p.ProbeContainer))
+	} else {
+		mux.HandleFunc(requestQueueHealthPath, healthState.HealthHandlerKProbe(p.ProbeContainer))
+	}
 	mux.HandleFunc(queue.RequestQueueDrainPath, healthState.DrainHandler())
 
 	return mux
 }
 
-func probeQueueHealthPath(port int, timeout time.Duration) error {
+func probeQueueHealthPath(port int, timeout int) error {
 	url := fmt.Sprintf(healthURLTemplate, port)
 
+	// use aggressive retries
+	if timeout == 0 {
+		return knativeProbe(url)
+	}
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			// Do not use the cached connection
+			DisableKeepAlives: true,
+		},
+		Timeout: time.Duration(timeout) * time.Second,
+	}
+
+	res, err := httpClient.Get(url)
+
+	if err != nil {
+		return errors.Wrap(err, "failed to probe")
```

Comment:
I'm not sure I understand this comment. Is the suggestion to include the URL in the error message? We build the URL on the fly, but (at present) it will always be `http://127.0.0.1:8022/health`.

Maybe something better than "failed to probe" is called for?

Reply:
Yep all I meant was `fail to probe: http://127.0.0.1:8022/health` provides a little bit more detail to someone looking at the error in the logs without having to trace through the code.